app/models/calculation_result.rb

```diff
@@ -26,6 +26,10 @@ def calculated?(section)
     api_response.dig(:result_summary, section, :proceeding_types).none? { _1[:result] == "pending" }
   end
 
+  def ineligible?(section)
+    api_response.dig(:result_summary, section, :proceeding_types).all? { _1[:result] == "ineligible" }
```

Comment:
interesting! So would CFE ever return eligible for some sections and ineligible for others? And is the result always present for each section? I guess I'm wondering how this looks to the user and how CFE determines the change to "ineligible"...

Reply:
CFE will go section by section, and as soon as the result for a section is "ineligible" it will stop and return the other sections as "pending". So you might get gross income as "eligible", disposable income as "ineligible" and capital as "pending", with an overall result of "ineligible". We've got logic (see the method above this one) to skip any sections that are "pending" in the UI.

Reply:
Nice, yes I have just spotted the "pending" thing! So this ineligibility sentence will only ever appear once. Got it! Do we think this will change with the one shot API?

Reply:
As it stands the one-shot API still uses the same bit of logic that short-circuits the eligibility check as soon as any section is ineligible. Changing that behaviour would be a breaking change, and the promise for the one-shot API is that it has the same behaviour, just in a different form factor. So I think we're save to assume this will continue to work going forward.